src/Http/ServerRequestFactory.php

```diff
@@ -94,6 +90,12 @@ public static function fromGlobals(
         ]);
 
         $request = static::marshalBodyAndRequestMethod($parsedBody ?? $_POST, $request);
+        // This is required as `ServerRequest::scheme()` ignores the value of
```

Comment:
@ADmad I noticed this change broke my setup containing the HTTPSEnforcerMiddleware.

Up until now I was setting `$this->request->trustProxy = true` in AppController::beforeFilter but this seems to have no effect for the MiddlewareQueue?

Reply:
@swiffer The `process()` method of the middleware get's the request instance as argument so use it's `setTrustedProxies()` method there.

Reply:
Got it, would you accept a PR to allow setting an array of trustedProxies as part of the config of httpsenforcermiddleware?

Reply:
Please open a new issue where we can discuss that.

Reply:
#16897